### PR TITLE
Display a :lock: icon for users who signed in with 2FA

### PR DESCRIPTION
**Pull request by rfk**
This consumes the new id_token claims from https://github.com/mozilla/fxa-oauth-server/pull/530 to determine whether the user signed in with 2FA, and if they did so, displays a nice little :lock: icon next to their email address.

The bulk of the change here is actually in the second commit, which makes the OAuth flow use [/.well-known/openid-configuration](https://accounts.firefox.com/.well-known/openid-configuration) and consume the OpenID Connect id_token.  The actual figuring-out-2fa-status part them just becomes a matter of looking for the "acr" claim in the id_token.

@vladikoff this :lock: business is mostly for my own demo purposes, but what do you think about merging at least the OIDC refactor stuff back to mainline?  I think it's a nice cleanup overall, but I might have unintentionally broken something else that this repo was doing.

### CODE REVIEW

File: oauth.js [hunk 1]
```diff
@@ -16,91 +16,112 @@ function toQueryString(obj) {
...
-    oauthParams.action = action;
+    params.action = action;
   }
-
   if (email) {
-    oauthParams.email = email;
+    params.email = email;
   }
+  request.get({
+    uri: config.issuer_uri + '/.well-known/openid-configuration',
```
Comment (vladikoff) [inline oauth.js:60]: aww that's nice 👍

File: config-local.json [hunk 2]
```diff
@@ -1,17 +1,9 @@
...
   "client_secret": "b93ef8a8f3e553a430d7e5b904c6132b2722633af9f03128029201d24a97f2a8",
   "redirect_uri": "http://127.0.0.1:8080/api/oauth",
-  "auth_uri": "http://127.0.0.1:9010/v1/authorization",
-  "oauth_uri": "http://127.0.0.1:9010/v1",
-  "content_uri": "http://127.0.0.1:3030",
-  "profile_uri": "http://127.0.0.1:1111/v1",
-  "scopes": "profile",
-  "preverify_email_jku": "http://127.0.0.1:8080/.well-known/public-keys",
-
+  "issuer_uri": "https://rfkelly.dev.lcip.org",
```
Comment (vladikoff) [inline config-local.json]: @rfk what should the local value be here?

File: package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "connect-fonts-alegreyasans": "0.0.2",
     "connect-fonts-opensans": "0.0.5",
     "express": "3.5.x",
-    "fxa-crypto-utils": "vladikoff/fxa-crypto-utils",
```
Comment (vladikoff) [inline package.json:29]: 👍

**Review by vladikoff — APPROVED** [review]
r+ with comments, mostly need to figure out the local.json config

File: config-local.json (hunk 2 above)
Comment (rfk) [reply config-local.json]: Oh whoops, this must have snuck in accidentally...

**Comment by rfk** [comment]
>  I think it's a nice cleanup overall, but I might have unintentionally broken something
> else that this repo was doing.

@vladikoff revisiting this, I think it probably breaks the "oauth in an iframe" flow that this repo also seems to test.  Do we actually still support that in content-server, or should I go ahead and remove it from here while I'm refactoring?

**Comment by rfk** [comment]
>  I think it probably breaks the "oauth in an iframe" flow that this repo also seems to test

From mtg: this flow is no longer supporter, let's remove any mention of it from 123done for completeness.

**Comment by vladikoff** [comment]
needs rebase , per https://github.com/mozilla/123done/pull/167

**Comment by rfk** [comment]
Rebased, and removed references to "rfkelly.dev.lcip.org", so I think this is good to go.

**Pull request merged**